Cod3rsGrowth.Web/wwwroot/webapp/controller/Detalhes.controller.js

```diff
@@ -5,42 +5,73 @@ sap.ui.define([
     "../formatter",
     "sap/m/MessageBox",
     "sap/ui/model/json/JSONModel",
+    "sap/m/MessageToast"
 
-], function (Base, ClubeServico, JogadorServico, Formatter, MessageBox,JSONModel) {
+], function (Base, ClubeServico, JogadorServico, Formatter, MessageBox, JSONModel, MessageToast) {
 	"use strict";
 
-    const CLUBES = "clubes";
+    const BARRA = "/";
+    const CLUBE = "clube";
     const DETALHES = "detalhes";
-    const TEXTO_ERRO_FETCH_CLUBE = "Clube não encontrado";
     const ARGUMENTS = "arguments";
     const DESTINO_VOLTAR = 'clubes';
     const TITULO_ERRO = "Erro";
     const DESTINO_EDITAR = 'editar';
     const PERGUNTA = "Deseja excluir este clube?";
     const TITULO_CONFIRMAR = "Confirme";
-    const JOGADORES = "jogadores"
+    const JOGADORES = "jogadores";
+    const JOGADOR = "jogador";
+    const NOME = "nome";
+    const DATADENASCIMENTO = "dataDeNascimento";
+    const ALTURA = "altura";
+    const PESO = "peso";
+    const IDCLUBE = "idClube";
+    const INPUTNOME = "InputNomeJogador";
+    const CALENDARIOCRIAR = "CalendarioCriarJogador";
+    const INPUTALTURA = "InputAltura";
+    const CLUBECRIACAO = "ClubeCriacaoJogador";
+    const INPUTPESO = "InputPeso";
+    const VAZIO = "";
+    const ESTADO_NONE = "None";
+    const ESTADO_ERROR = "Error";
+    const TEXTO_ERRO_NOME = "Campo 'Nome' precisa ser preenchido.";
+    const TEXTO_ERRO_ALTURA = "Campo 'Altura' precisa ser preenchido.";
+    const TEXTO_ERRO_NASCIMENTO = "Campo 'Data de Nascimento' precisa ser preenchido.";
+    const TEXTO_ERRO_CLUBE = "Campo 'Clube' precisa ser preenchido.";
+    const TEXTO_ERRO_PESO = "Campo 'Peso' precisa ser preenchido.";
+    const MENSAGEM_SUCESSO_JOGADOR = "Jogador criado com sucesso!";
+    const MENSAGEM_SUCESSO_JOGADOR_EDITADO = "Jogador editado com sucesso!";
+    const INDICEUM = 1;
+    const EDITAR = "editar";
+    const DURACAO_TOAST = 5000;
+    const MENSAGEM_ERRO_DESCONHECIDO = "Erro desconhecido encontrado!";
+    const DETALHES_ERRO_INDISPONIVEL = "Stacktrace está indisponível!";
+    const QUEBRADELINHA = "\r\n";
+    const IDADE = "idade";
 
 	return Base.extend("cod3rsgrowth.webapp.controller.Detalhes", {
         formatter: Formatter,
         clubeServico : ClubeServico,
         jogadorServico : JogadorServico,
 
         onInit: function () {
+            this.DadosCriacao = [];
             this._vincularRota(DETALHES, this.aoCoincidirRota);
         },
         aoCoincidirRota : async function(evento){
+            this.carregarModelo();
             const argumentos = evento.getParameter(ARGUMENTS);
 
             await this.clubeServico.aoBuscarClubePorId(argumentos.idClube)
             .then((clube) => {
                 var oModel = new JSONModel(clube);
-                this.getView().setModel(oModel, CLUBES);
+                this.getView().setModel(oModel, CLUBE);
```

Comment:
esses this.getView().setModel e this.getView().getModel esta sendo repetido em varios lugares cria uma funcao pra fazer isso